lib/tool_shed/webapp/buildapp.py

```diff
@@ -36,6 +36,10 @@ class ToolShedGalaxyWebTransaction(GalaxyWebTransaction):
     def repositories_hostname(self) -> str:
         return url_for("/", qualified=True).rstrip("/")
 
+    def get_or_create_default_history(self):
+        # tool shed has no concept of histories
+        return None
```

Comment:
This may complicate type annotation when we add it. What about guarding the change in https://github.com/galaxyproject/galaxy/pull/17557 with a `if self.webapp.name == "galaxy":` instead?

Reply:
You can raise `NotImplementedError`, I suppose?